service/controller/v12/resource/instance/create.go

```diff
@@ -187,9 +188,36 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	if hasResourceStatus(customObject, Stage, ProvisioningSuccessful) {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "vmss deployment successful")
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting resource status to '%s/%s'", Stage, ScaleVMSSToDouble))
+
+		err := r.setResourceStatus(customObject, Stage, ScaleVMSSToDouble)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("set resource status to '%s/%s'", Stage, ScaleVMSSToDouble))
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
+		reconciliationcanceledcontext.SetCanceled(ctx)
+		return nil
+	}
+
+	if hasResourceStatus(customObject, Stage, ScaleVMSSToDouble) {
+		// Double the desired number of nodes in worker VMSS in order to
+		// provide 1:1 mapping between new up-to-date nodes when draining and
+		// terminating old nodes.
+		nodeCountFunc := func(_ int64) int64 {
+			return int64(key.WorkerCount(customObject) * 2)
+		}
+
+		err := r.scaleVMSS(ctx, customObject, key.WorkerVMSSName, nodeCountFunc)
```

Comment:
I don't get why you are passing a function instead of the computed int64 here.

Reply:
My original implementation was following:
```go
nodeCountFunc := func(n int64) int64 {
    return n * 2
}
```

...but then I realized that in case the operator crashes during `InstancesUpgrading`, it would in the worst case double the size of VMSS on every iteration. Then I settled with the const value there, but I felt that the function signature is more flexible with the function. That way implementation can avoid reading of `VMSS` state several times during operation.

Reply:
ok